samples/bluetooth/fast_pair/locator_tag/include/app_factory_reset.h

```diff
@@ -20,16 +20,31 @@
 extern "C" {
 #endif
 
-/** Callback type used to notify the user that the factory reset has been executed. */
-typedef void (*app_factory_reset_executed_cb)(void);
+struct app_factory_reset_callbacks {
+	/** Callback used to allow the user to prepare to the factory reset. */
+	void (*prepare)(void);
+
+	/** Callback used to notify the user that the factory reset has been executed. */
+	void (*executed)(void);
+};
+
+/** Register a factory reset callbacks.
+ *
+ *  @param _name Factory reset callbacks structure name.
+ *  @param _prepare Callback to define actions before the factory reset.
+ *  @param _executed Callback to define actions after the factory reset.
+ */
+#define APP_FACTORY_RESET_CALLBACKS_REGISTER(_name, _prepare, _executed)	\
+	static const STRUCT_SECTION_ITERABLE(app_factory_reset_callbacks, _name) = {	\
+		.prepare = _prepare,						\
+		.executed = _executed,						\
+	}
 
 /** Schedule the factory reset action.
  *
  * @param delay Time to wait before the factory reset action is performed.
- * @param cb    Callback to define the prepare action before the factory reset.
  */
-void app_factory_reset_schedule(k_timeout_t delay,
-				app_factory_reset_executed_cb cb);
+void app_factory_reset_schedule(k_timeout_t delay);
```

Comment:
Consider returning an error when this one fails (it may be reasonable looking at newly introduced architecture changes)

Reply:
Similar for `cancel`

Reply:
IMO currently there is no need for it, I won't add it.

Reply:
Generally upper layer should be aware that something went wrong (logging error may not be enough here I think). Since it used to be like that I would not insist on modifying it in scope of this PR, but generally it may need to be updated at some point.

Reply:
Added as a part of the follow-up: https://nordicsemi.atlassian.net/browse/NCSDK-28552